lib/util/host-rules.ts

```diff
@@ -23,7 +22,7 @@ export function add(params: HostRule): void {
   let resolvedHost =
     params.baseUrl || params.hostName || params.domainName || params.matchHost;
   if (resolvedHost) {
-    resolvedHost = URL.parse(resolvedHost).hostname || resolvedHost;
+    resolvedHost = parseUrl(resolvedHost)?.hostname || resolvedHost;
```

Comment:
This would be a problem for our `isUrl()` function if so. It's logic currently is:

```
function isUrl(url: string): boolean {
  return !!parseUrl(url)?.hostname;
}
```